src/test/java/core/basesyntax/services/fileprocessing/TransactionGetterImplTest.java

```diff
@@ -0,0 +1,122 @@
+    }
+
+    @Test
+    void getTransactionsData_putBadDataWithNumbersInWords_notOk() {
+        RuntimeException numberFormatException = assertThrows(RuntimeException.class,
+                () -> transactionGetter.getTransactionsData(BAD_INTEGERS));
+        assertTrue(numberFormatException.getMessage().startsWith(EXPECTED_EXCEPTION_MESSAGE));
+    }
+
+    @Test
+    void getTransactionsData_putBadDataWithNonExistingOperations_notOk() {
+        assertThrows(OperationException.class,
+                () -> transactionGetter.getTransactionsData(BAD_OPERATIONS));
+    }
+
+    @Test
+    void getTransactionsData_putNormalButDifferentData() {
+        List<FruitTransaction> normalFruitTransactions =
+                fruitTransactionsTestSetter(NORMAL_TEST_DATA);
+        List<FruitTransaction> anotherNormalFruitTransactions =
+                fruitTransactionsTestSetter(ANOTHER_NORMAL_TEST_DATA);
+        assertFalse(assertListEquals(normalFruitTransactions, anotherNormalFruitTransactions));
+        assertFalse(assertListEquals(normalFruitTransactions,
+                transactionGetter.getTransactionsData(ANOTHER_NORMAL_TEST_DATA)));
+        assertFalse(assertListEquals(transactionGetter.getTransactionsData(NORMAL_TEST_DATA),
+                transactionGetter.getTransactionsData(ANOTHER_NORMAL_TEST_DATA)));
+    }
+
+    @Test
+    void getTransactionsData_putNormalData_Ok() {
+        List<FruitTransaction> normalFruitTransactions =
+                fruitTransactionsTestSetter(NORMAL_TEST_DATA);
+        assertTrue(assertListEquals(normalFruitTransactions,
+                transactionGetter.getTransactionsData(NORMAL_TEST_DATA)));
+    }
+
+    private List<FruitTransaction> fruitTransactionsTestSetter(List<String[]> testData) {
+        List<FruitTransaction> fruitTransactions = new ArrayList<>();
+        for (String[] strings : testData) {
+            FruitTransaction fruitTransaction = new FruitTransaction();
+            fruitTransaction.setOperation(
+                    FruitTransaction.Operation.getOperationFromCode(strings[OPERATION_INDEX]));
+            fruitTransaction.setFruit(strings[FRUIT_INDEX]);
+            fruitTransaction.setQuantity(Integer.parseInt(strings[QUANTITY_INDEX]));
+            fruitTransactions.add(fruitTransaction);
+        }
+        return fruitTransactions;
+    }
+
+    private boolean assertListEquals(List<FruitTransaction> expectedResult,
+                                     List<FruitTransaction> actualResult) {
+        for (int i = 0; i < expectedResult.size(); i++) {
+            if (!expectedResult.get(i).getOperation().equals(actualResult.get(i).getOperation())
+                    || !expectedResult.get(i).getFruit().equals(actualResult.get(i).getFruit())
+                    || expectedResult.get(i).getQuantity() != actualResult.get(i).getQuantity()) {
+                return false;
+            }
+        }
+        return true;
+    }
```

Comment:
have you tried to you override equals and hashcode for fruitTransaction and use JUnit's standard equality assertion?

Reply:
I did think about that actually, I just thought I am not allowed to change base code. Now that I know I am, I'll do it :)